parser/build.rs

```diff
@@ -2,15 +2,19 @@ use std::fmt::Write as _;
 use std::fs::File;
 use std::io::{self, BufRead, BufReader, BufWriter, Write};
 use std::path::PathBuf;
-use std::process::{self, Command};
+use std::process::{Command, ExitCode};
 use tiny_keccak::{Hasher, Sha3};
 
-fn main() {
-    check_lalrpop("src/python.lalrpop", "src/python.rs");
+fn main() -> ExitCode {
+    let result = check_lalrpop("src/python.lalrpop", "src/python.rs");
+    if let Some(result) = result {
+        return result;
+    }
     gen_phf();
+    ExitCode::SUCCESS
 }
 
-fn check_lalrpop(source: &str, generated: &str) {
+fn check_lalrpop(source: &str, generated: &str) -> Option<ExitCode> {
```

Comment:
I think this function is expected to return Result instead of Option<ExitCode>

Reply:
There is a control statement that returns `None` in the same function, so isn't it possible to reach that code?

Reply:
Now, error case is encoded as `Some(ExitCode)` and successful case is encoded as `None`. This is typically encoded as Result type in Rust code like `Ok(())` and `Err(_)`. Give please give a proper error to `Err` case.

Reply:
Edit: because error type is not what this PR covers, `Result<(), ExitCode>` will be good enough.